diff --git a/arch/arm/src/armv8-m/arm_doirq.c b/arch/arm/src/armv8-m/arm_doirq.c
index 6cd723df34e..a28c58b74dd 100644
--- a/arch/arm/src/armv8-m/arm_doirq.c
+++ b/arch/arm/src/armv8-m/arm_doirq.c
@@ -52,10 +52,10 @@ void exception_direct(void)
 #ifdef CONFIG_ARCH_FPU
   __asm__ __volatile__
     (
-      "mov r0, %0\n"
-      "vmsr fpscr, r0\n"
+      "vmsr fpscr, %0\n"
       :
-      : "i" (ARM_FPSCR_LTPSIZE_NONE)
+      : "r" (ARM_FPSCR_LTPSIZE_NONE)
+      : "memory"
     );
 #endif
 
